app/controllers/assets_controller.rb

```diff
@@ -13,22 +13,14 @@ def update
 
   def destroy
     @asset = find_asset
-
-    if @asset.destroy
-      render json: AssetPresenter.new(@asset, view_context).as_json(status: :success)
-    else
-      error 422, @asset.errors.full_messages
-    end
+    @asset.destroy
```

Comment:
I saw from the commit message about switching this to a 500 which makes sense, what I wondered is whether we're confident this definitely produces a 500?

If this was ActiveRecord the `ActiveRecord::RecordInvalid` exception would be caught and converted to a 422 by action_dispatch.rescue_responses by default and it looks like Mongoid applies the same in a Railtie: https://github.com/mongodb/mongoid/blob/3cbbcfb42b091261f7f255959c48d98a1b5948cb/lib/mongoid/railtie.rb#L26

Reply:
That's true. I was naively assuming that most records would be valid, and the text in the (now) deleted test suggested the developer was more concerned about a scenario where the DB operation itself failed, although admittedly that wouldn't populate `errors`!

How about I rephrase the commit to be more: it'll do the right thing without us testing it?

Reply:
Oh so sorry for the long delay.

The commit message change sounds reasonable, every other route I consider seems to end up in massive rabbit hole.  It may mean a client gets a 422 in an unexpected format, but then no-one can do anything about a 422 here as it's the system state that is wrong (and thus the rabbit hole).